From 4730a3b164ac45a6e5b48b2484dd72028982af56 Mon Sep 17 00:00:00 2001
From: Dylan Van Assche <me@dylanvanassche.be>
Date: Sun, 26 Dec 2021 11:35:55 +0100
Subject: [PATCH 1/2] plugins/quectel/mm-broadband-modem-qmi-quectel: ignore
 carrier revision

Quectel EG25-G QMI modem firmware revisions are not matched with carrier configs.
Different VoLTE profiles might be selected by the user causing
the carrier config revision to change.

Since fwupd uses this string for firmware updates, it gets confused and
thinks the user has an update available.
---
 plugins/quectel/mm-broadband-modem-qmi-quectel.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/plugins/quectel/mm-broadband-modem-qmi-quectel.c b/plugins/quectel/mm-broadband-modem-qmi-quectel.c
index 8cb290c5..676006f4 100644
--- a/plugins/quectel/mm-broadband-modem-qmi-quectel.c
+++ b/plugins/quectel/mm-broadband-modem-qmi-quectel.c
@@ -52,6 +52,8 @@ mm_broadband_modem_qmi_quectel_new (const gchar  *device,
                          MM_BASE_MODEM_PLUGIN, plugin,
                          MM_BASE_MODEM_VENDOR_ID, vendor_id,
                          MM_BASE_MODEM_PRODUCT_ID, product_id,
+                         /* exclude carrier information */
+                         MM_IFACE_MODEM_FIRMWARE_IGNORE_CARRIER, TRUE,
                          /* QMI bearer supports NET only */
                          MM_BASE_MODEM_DATA_NET_SUPPORTED, TRUE,
                          MM_BASE_MODEM_DATA_TTY_SUPPORTED, FALSE,
-- 
2.33.1

